pkg/api/git.go

```diff
@@ -0,0 +1,63 @@
+package api
+
+import (
+	"github.com/avorty/spito/pkg/shared"
+	"github.com/avorty/spito/pkg/vrct/vrctFs"
+	"github.com/go-git/go-git/v5"
+	"os"
+	"path/filepath"
+)
+
+type GitApi struct {
+	FsVrct *vrctFs.VRCTFs
+}
+
+func (g *GitApi) GitClone(repoUrl, destinationPath string) error {
+	if err := os.MkdirAll("/tmp/spito", shared.DirectoryPermissions); err != nil {
+		return err
+	}
+
+	tmpDir, err := os.MkdirTemp("/tmp/spito", "git-clone-tmp-")
+	if err != nil {
+		return err
+	}
+
+	if _, err = git.PlainClone(tmpDir, false, &git.CloneOptions{URL: repoUrl}); err != nil {
+		return err
+	}
+
+	if err := moveRecursively(g.FsVrct, tmpDir, destinationPath); err != nil {
+		return err
+	}
+
+	return os.RemoveAll(tmpDir)
+}
+
+func moveRecursively(fsVrct *vrctFs.VRCTFs, from, to string) error {
```

Comment:
I would move this function to vrct

Reply:
1) it is temporary solution, there should be other way in vrct to move files
2) it just uses vrct, there's no part of that code which is specific for vrct

Reply:
It is specific for VRCT.. It's kind of wrapper or helper around VRCT that can be outside `git.go`